plugins/catalog-backend/src/processing/DefaultCatalogProcessingEngine.test.ts

```diff
@@ -319,4 +335,115 @@ describe('DefaultCatalogProcessingEngine', () => {
 
     await engine.stop();
   });
+
+  it('should stitch both the previous and new sources when relations change', async () => {
+    const engine = new DefaultCatalogProcessingEngine(
```

Comment:
note to self: blech, all of the tests in this file mock WAY too much and should work against a test database instead